core/src/main/scala/org/http4s/headers/Host.scala

```diff
@@ -17,14 +17,31 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.parse.Parser
+import org.http4s.internal.parsing.Rfc3986
 import org.http4s.util.Writer
+import scala.util.Try
 
 object Host extends HeaderKey.Internal[Host] with HeaderKey.Singleton {
   def apply(host: String, port: Int): Host = apply(host, Some(port))
 
   override def parse(s: String): ParseResult[Host] =
-    HttpHeaderParser.HOST(s)
+    ParseResult.fromParser(parser, "Invalid Host")(s)
+
+  //  // Do not accept scoped IPv6 addresses as they should not appear in the Host header,
+  //  // see also https://issues.apache.org/bugzilla/show_bug.cgi?id=35122 (WONTFIX in Apache 2 issue) and
+  //  // https://bugzilla.mozilla.org/show_bug.cgi?id=464162 (FIXED in mozilla)
+  private[http4s] val parser = {
+    val host =
+      (Uri.Ipv4Address.parser).orElse(Uri.RegName.parser)
+    val port = Parser.string(":") *> Rfc3986.digit.rep.string.mapFilter { s =>
+      Try(s.toInt).toOption
```

Comment:
A horrid thing about this spec is that the port can be unbounded.  I've been thinking about this in the URI redesign, as well.  I don't think it's _useful_, but it stands in the way of compliance.

I think this is fine for now, but I'm trying to think of a way out that's both practical and pedantic.